app/components/member/Edit/role.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Relay from 'react-relay';
+
+import Button from '../../shared/Button';
+import FormCheckbox from '../../shared/FormCheckbox';
+import Panel from '../../shared/Panel';
+
+import FlashesStore from '../../../stores/FlashesStore';
+
+import OrganizationMemberUpdateMutation from '../../../mutations/OrganizationMemberUpdate';
+
+import OrganizationMemberRoleConstants from '../../../constants/OrganizationMemberRoleConstants';
+
+class Role extends React.PureComponent {
+  static displayName = "Member.Edit.Role";
+
+  static propTypes = {
+    organizationMember: PropTypes.shape({
+      role: PropTypes.string.isRequired,
+      permissions: PropTypes.object.isRequired,
+      user: PropTypes.shape({
+        name: PropTypes.string.isRequired
+      }).isRequired
+    }),
+    isSelf: PropTypes.bool.isRequired
+  };
+
+  state = {
+    isAdmin: false
+  };
+
+  componentWillMount() {
+    this.setState({
+      isAdmin: this.props.organizationMember.role === OrganizationMemberRoleConstants.ADMIN
+    });
+  }
+
+  render() {
+    let content;
+
+    // Don't show the remove panel if you can't actually update them
+    if (!this.props.organizationMember.permissions.organizationMemberUpdate.allowed) {
+      content = (
+        <Panel>
+          <Panel.Row>
+            {this.props.organizationMember.permissions.organizationMemberUpdate.message}
+          </Panel.Row>
+        </Panel>
+      );
+    } else {
+      const saveRowContent = (
+        this.props.isSelf
+          ? <span className="dark-gray">You can’t edit your own roles</span>
+          : (
+            <Button
+              onClick={this.handleUpdateOrganizationMemberClick}
+              loading={this.state.updating && 'Saving…'}
+            >
+              Save
+            </Button>
+          )
+      );
```

Comment:
Yeah, I think when I first built this I hadn’t realised that there was a permission with per-object details!

That said, the server-provided message is a bit ugly and awkward, so it might be worth changing.